daphne/src/lib.rs

```diff
@@ -1173,7 +1178,9 @@ impl<S> DapRequest<S> {
     /// the version in use is not draft02. If the caller is not the Collector, or draft02 is in
     /// use, we exepct the collection job ID to be missing.
     pub fn collection_job_id(&self) -> Result<&CollectionJobId, DapAbort> {
-        if let DapResource::CollectionJob(ref collection_job_id) = self.resource {
+        if let DapResource::CollectionJob(collection_job_id) = &self.resource {
+            Ok(collection_job_id)
+        } else if let Some(collection_job_id) = &self.collection_job_id {
```

Comment:
Why make this change? There's no explanation in the commit message.

Why leave the `CollectionJob` variant of `DapResource` intact?

Note: `DapResource` is meant to encapsulate an idea from the DAP spec, namely that a "collection job" is a "resource". If there's a practical reason to delete this, I think it's fine, but I don't understand why we wouldn't also change `DapResource` in the same way.